src/modules/transaction/usecase/service/transaction.service.ts

```diff
@@ -33,13 +37,29 @@ export class TransactionService {
         await this.transactionRepository.deleteTransactionBy(transactionId);
     }
 
+    async computeUserBalance(authUserId: string): Promise<number> {
+        const transactions: TransactionOutput[] = await this.transactionRepository.getTransactions(authUserId);
+        const userBalance: number = transactions.reduce((acc, transaction) => {
```

Comment:
@PCalape Check if there is a way to compute the sum directly using Supabase. In SQL you can get the sum directly , maybe Supabase supports that as well so we don't have to compute it ourselves.

Reply:
Hello @arnellebalane will check this out thank you!

Reply:
Hello @arnellebalane upon checking with the docs and other resources, it seems like supabase does not support aggregations for now except for count

Reply:
Hello @arnellebalane, will explore postgres functions in supabase for this one. Thanks!

Reply:
@PCalape Okay this is fine for now 👍

Reply:
Hello @arnellebalane, kindly review again as I found a way to make the computation of user balance a database operation using postgres function. Thank you!